### ISSUE TITLE
Does Trivy support scanning for dotnet?  

### ISSUE THREAD

**Issue by daxin09pp** [aquasecurity/trivy#1390]
I used Trivy Image to scan dotnet Image, which should have the CVE of [CVE-2021-41355](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-41355), but Trivy did not report it.

The command I used was:
`docker run --rm -v /var/run/docker.sock:/var/run/docker.sock     -v $HOME/Library/Caches:/root/.cache/ aquasec/trivy mcr.microsoft.com/dotnet/aspnet:5.0.10-alpine3.14`

But Microsoft Defender for Container Registries can report this CVE.  

Do you have any good suggestions or am i using it wrong?

**Comment by afdesk** [comment]
@daxin09pp thanks for your report!
I'll try to investigate it

**Comment by afdesk** [comment]
Hi @daxin09pp!

`trivy` looks for vulnerabilities of OS packages (Alpine, RHEL, CentOS, etc.) and language-specific packages (Bundler, Composer, npm, yarn, etc.).

Your image's OS is alpine and `trivy` can detect 22 vulnerabilities.

for dotnet `trivy` supports scanning only `packages.lock.json` yet.

it seems that your case isn't in scope of the current features.

**Comment by daxin09pp** [comment]
Hi @afdesk ,
Do you have any plans to support my case?  .Net core is hot right now, and a lot of people are using it.  I think supporting it would be great for Trivy.  
Thanks,

**Comment by afdesk** [comment]
Hi @daxin09pp! thanks for interest in trivy.

at this time there is no plan to support scanning basic .Net images. 
but `trivy` can look for vulnerabilities inside specific .Net projects (checking `packages.lock.json`) and when .NET itself is installed by package managers.

**Comment by daxin09pp** [comment]
Hi @afdesk , Thanks for your reply!
At present, most images using .Net Core are built based on Microsoft's official image. It is hoped that Trivy can support such image Scan.  
Thanks!

**Comment by olljanat** [comment]
> Do you have any good suggestions or am i using it wrong?

You need enable `RestorePackagesWithLockFile` like it is described on [here](https://devblogs.microsoft.com/nuget/enable-repeatable-package-restores-using-a-lock-file/) to get `packages.lock.json` file which is needed by Trivy scan.

FYI. I just created this PR https://github.com/dotnet/dotnet-docker/pull/3335 about including Trivy to official samples. It is probably good idea to give couple of 👍 for it so it gets more likely to merged.

**Comment by olljanat** [comment]
I just noticed that there is two different things on this discussion. With package lock file Trivy can detect vulnerabilities in packages which are used by .NET project. However it does not detect vulnerabilities in .NET it selves which is actually the case which was asked on first message.

Tested with `trivy image mcr.microsoft.com/dotnet/runtime:6.0.5` and it does not list *CVE-2022-30184* which [was fixed by 6.0.6](https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0.6/6.0.6.md). However that vulnerability can be found from [Red Hat CVE database](https://access.redhat.com/security/cve/cve-2022-30184) which [based on documentation](https://aquasecurity.github.io/trivy/v0.29.2/community/contribute/issue/#red-hat-cve-database) is one of the sources.

> Hey, syft just added support for scanning dot-net without the need for generating lock files, just saying...

True. That implementation can be found from https://github.com/anchore/grype/pull/747 if someone is interested to add it to Trivy too. However even that does not detect vulnerabilities in .NET it selves  https://github.com/anchore/grype/issues/736#issuecomment-1166576521

**Comment by knqyf263** [comment]
FYI: We're also working on it.
https://github.com/aquasecurity/fanal/pull/550

**Comment by olljanat** [comment]
@knqyf263 @astevenson-microsoft that repository looks to be archived so new pull request should be created to here.

**Comment by ckotzbauer** [comment]
> Tested with trivy image mcr.microsoft.com/dotnet/runtime:6.0.5 and it does not list https://github.com/advisories/GHSA-3885-8gqc-3wpf which [was fixed by 6.0.6](https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0.6/6.0.6.md). However that vulnerability can be found from [Red Hat CVE database](https://access.redhat.com/security/cve/cve-2022-30184) which [based on documentation](https://aquasecurity.github.io/trivy/v0.29.2/community/contribute/issue/#red-hat-cve-database) is one of the sources.

The vulnerabilities listed in the CVE (Nuget.* packages) are not part of the dotnet-runtime image. The result is correct.

**Comment by iamNoah1** [comment]
> Hey, syft just added support for scanning dot-net without the need for generating lock files, just saying...

I can confirm that this is not the case. Only after adding lock files, security issues have been reported for my .net6 application.

**Comment by knqyf263** [comment]
`.deps.json` support is out.
https://github.com/aquasecurity/trivy/pull/2487

**Comment by jkone27** [comment]
we get dotnet-core (deps.json) scans, on new NET5 and NET6 projects which have no .deps.json , and we received vulnerabilities from 2019 in 2022 NET6 images. how should we adress this? where do you pass the dotnet version hint to trivy to not make this kind of mistakes?

**Comment by thiago-gitlab** [comment]
@knqyf263, you might be interested in reading through https://gitlab.com/gitlab-org/gitlab/-/issues/372402. TL;DR:

> the .deps.json file isn't the right choice for container scanning
